torch/amp/autocast_mode.py

```diff
@@ -273,96 +271,53 @@ def __init__(
-            if self.fast_dtype not in supported_dtypes:
-                error_message = "In IPU autocast, but the target dtype is not supported. Disabling autocast.\n"
-                error_message += "IPU Autocast only supports dtypes of torch.bfloat16 and torch.float16 currently."
-                warnings.warn(error_message)
-                enabled = False
-        elif self.device == "hpu":
-            supported_dtype = [torch.bfloat16, torch.float16]
-            if self.fast_dtype not in supported_dtype:
-                error_message = "In HPU autocast, but the target dtype is not supported. Disabling autocast.\n"
-                error_message += "HPU Autocast only supports dtypes of torch.bfloat16 and torch.float16 currently."
-                warnings.warn(error_message)
-                enabled = False
-        elif self.device == self.custom_backend_name:
-            supported_dtype = self.custom_device_mod.get_amp_supported_dtype()
-            if self.fast_dtype not in supported_dtype:
-                error_message = f"In {self.custom_backend_name} autocast, but the target dtype {self.fast_dtype} is not supported. "
-                error_message += f"Disabling autocast.\n {self.custom_backend_name} Autocast only supports dtypes of "
-                error_message += (
-                    ", ".join(str(dtype) for dtype in supported_dtype) + " currently."
-                )
-                warnings.warn(error_message)
-                enabled = False
-        elif self.device == "cuda":
-            if (
-                enabled
-                and self.fast_dtype == torch.bfloat16
-                and not torch.cuda.is_bf16_supported()
-            ):
-                raise RuntimeError(
-                    "Current CUDA Device does not support bfloat16. Please switch dtype to float16."
-                )
-        elif self.device == "mps":
-            supported_dtype = [torch.bfloat16, torch.float16]
-            if self.fast_dtype not in supported_dtype:
+        device_supported_dtypes = {
+            "cpu": [torch.bfloat16, torch.float16],
+            "mtia": [torch.bfloat16, torch.float16],
+            "maia": [torch.bfloat16, torch.float16],
+            "xpu": [torch.bfloat16, torch.float16],
+            "ipu": [torch.bfloat16, torch.float16],
+            "hpu": [torch.bfloat16, torch.float16],
+            "mps": [torch.bfloat16, torch.float16],
+            "xla": [torch.bfloat16, torch.float16],
+            self.custom_backend_name: self.custom_device_mod.get_amp_supported_dtype(),
+        }
+
+        if self.device in device_supported_dtypes:
+            supported_dtypes = device_supported_dtypes[self.device]
+            device_name = (
+                self.device
+                if self.device == self.custom_backend_name
+                else self.device.upper()
+            )
+
+            if self.fast_dtype not in supported_dtypes and enabled:
```

Comment:
```suggestion
            if self.fast_dtype not in supported_dtypes and enabled:
```
This is the same if as in line 278 but every other instance does not have the `and enabled`. That appears to be different from the original behavior, yes?